pymongo/pool.py

```diff
@@ -571,7 +572,9 @@ def hello_cmd(self):
         if self.opts.server_api:
             return SON([(HelloCompat.CMD, 1)])
         else:
-            return SON([(HelloCompat.LEGACY_CMD, 1)])
+            if self.hello_ok:
+                return SON([(HelloCompat.CMD, 1)])
```

Comment:
Suggest merging this with the server_api check above:
```
if self.opts.server_api or self.hello_ok:
...
```

Reply:
Not needed anymore since `helloOk` is only sent on initial handshake.